### PR TITLE
fix: OkHttpClientFactory.additionalConfig can be used to override the default OkHttp Dispatcher

### LINKED ISSUE #4797: Restore ability to make OkHttp's `Dispatcher` use daemon threads
### Is your enhancement related to a problem? Please describe

Up to version 6.3.1, it was possible to pass a custom `ExecutorService` to OkHttp client's `Dispatcher`, thanks to `OkHttpClientFactory`:
```java
  /**
   * Subclasses may use this to apply additional configuration after the Config has been applied
   * This method is only called for clients constructed using the Config.
   *
   * @param builder
   */
  protected void additionalConfig(OkHttpClient.Builder builder) {
    // no default implementation
  }
```
... which used to allow, for instance, to make sure the thread pool spawns _daemon threads_:
```java
            @Override
            protected void additionalConfig(final Builder builder) {
                builder.dispatcher(new Dispatcher(Executors.newCachedThreadPool(
                    new ThreadFactoryBuilder().setDaemon(true).setNameFormat("k8s-client-%d").build())));
            }
```

Since version 6.4.0, due to "Test/simultaneous HttpClient connections" (#4687), the default `Dispatcher` creation happens _after_ `additionalConfig`, thus disregarding any user-provided `ExecutorService`.

### Describe the solution you'd like

To get the best of both worlds, i.e. both the default `Dispatcher` configuration from #4687 and a daemon thread pool:
1. either always set a cached thread pool spawning daemon threads when configuring the default `Dispatcher` (assuming this forms a consensus, of course),
2. or allow to pass a custom `ExecutorService` at some stage that would be later used when configuring the default `Dispatcher`, for instance leveraging _existing_ `KubernetesClientBuilder.withTaskExecutor[Supplier]` (probably a good option),
3. or mutate the builder's `Dispatcher` in place (obtained through `builder.dispatcher()`),
4. or invoke `additionalConfig` _after_ having configured the default `Dispatcher`. In this case users would have to replicate any default configuration by themselves (through `[get|set]IdleCallback`, `[get|set]MaxRequests` and `[get|set]MaxRequestsPerHost` as of now).

### Describe alternatives you've considered

Something as arguable as:
```java
    private static KubernetesClient replaceOkHttpThreadFactory(final KubernetesClient client) {
        final var httpClient = (OkHttpClientImpl) client.getHttpClient();
        final var pool = (ThreadPoolExecutor) httpClient.getOkHttpClient().dispatcher().executorService();
        pool.setThreadFactory(new ThreadFactoryBuilder().setDaemon(true).setNameFormat("k8s-client-%d").build());
        return client;
    }
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by manusa**
## Description

Fix #4797 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: httpclient-okhttp/src/test/java/io/fabric8/kubernetes/client/okhttp/OkHttpClientFactoryTest.java [hunk 1]
```diff
@@ -53,4 +56,21 @@ void shouldNotRespectMaxRequestsPerHost() {
     assertEquals(Integer.MAX_VALUE, client.getOkHttpClient().dispatcher().getMaxRequestsPerHost());
   }
 
+  @Test
+  void additionalConfigCanOverrideDispatcher() {
+    final Dispatcher dispatcher = new Dispatcher();
```
Comment (shawkins) [inline httpclient-okhttp/src/test/java/io/fabric8/kubernetes/client/okhttp/OkHttpClientFactoryTest.java:61]: Overriding completely via additionalConfig does preserve the legacy behavior.  Not sure if it needs an explicit test, but the intent of #4797 would be to override the initDispatcher:

> To get the best of both worlds, i.e. both the default Dispatcher configuration from https://github.com/fabric8io/kubernetes-client/pull/4687 and a daemon thread pool

Comment (manusa) [reply httpclient-okhttp/src/test/java/io/fabric8/kubernetes/client/okhttp/OkHttpClientFactoryTest.java:61]: I didn't add a test for the initDispatcher override, but this fix allows for both. You can either override initDispatcher, or use additionalConfig.

I can add that test too. Not sure if something else is expected besides that

**Review by rohanKanojia — APPROVED** [review]

**Review by shawkins — APPROVED** [review]
LGTM

**Comment by rdesgroppes** [comment]
Thanks!

**Pull request merged**